lib/collection.js

```diff
@@ -378,14 +378,14 @@ Collection.prototype.find = function(query, options, callback) {
   // Decorate find command with collation options
   decorateWithCollation(findCommand, this, options);
 
-  // Create the cursor
-  if (typeof callback === 'function')
-    return handleCallback(
-      callback,
-      null,
-      this.s.topology.cursor(this.s.namespace, findCommand, newOptions)
-    );
-  return this.s.topology.cursor(this.s.namespace, findCommand, newOptions);
+  const cursor = this.s.topology.cursor(this.s.namespace, findCommand, newOptions);
+
+  // add a map function to the cursor if "deserialize" option is set
+  if (this.s.options.deserialize && typeof this.s.options.deserialize === 'function') {
```

Comment:
this `if` condition can be condensed into just `typeof this.s.options.deserialize === 'function'`.